Add LSP-robloxlua

alihsaas opened the pull request.

alihsaas committed:
Add LSP-robloxlua

rchl committed:
Merge branch 'main' into main

rwols commented:
I think we should wait with merging this. I have concocted a method to put a built zipped package in a draft release. This is much cleaner than committing all of the binary artifacts in the git repository.

rwols commented:
A blocker for my method is that Package Control doesn't handle redirects (https://github.com/wbond/package_control/issues/1502). The URL to the zip files are redirected by GitHub (with a HTTP 302 response) to some amazon S3 storage somewhere on this planet.

To see how it can work, see https://github.com/sublimelsp/lsp-lua/
It is entirely based on "github workflows".

1) There is a github workflow that I trigger manually from the "Actions" tab on the repo: https://github.com/sublimelsp/LSP-lua/blob/main/.github/workflows/build-draft-release.yml this creates a draft release and uploads three zip files, ready to be unzipped by Package Control
2) I browse to the draft release and then click on "publish release"
3) This triggers another github workflow: https://github.com/sublimelsp/LSP-lua/blob/main/.github/workflows/notify-repository-on-published-release.yml this workflow makes a HTTP REST api call to **this** github repo
4) **this** github repo has a workflow that awaits a rest call and then automatically creates a pull request from it: https://github.com/sublimelsp/repository/blob/main/.github/workflows/on-new-release.yml

Writing this out, I am wondering whether a third party can also do the HTTP REST call and trigger a pull request automatically...

alihsaas commented:
Alright, thanks, I will close this PR then and generate a new one once the LSP is ready.

rwols commented:
Let’s keep this open otherwise I’ll forget about it

alihsaas committed:
Add LSP-robloxluau

rwols commented:
I remember there were XML files in the package previously, but now I don't see them in the zip files. Is this intended?

alihsaas commented:
> 
> 
> I remember there were XML files in the package previously, but now I don't see them in the zip files. Is this intended?
Yah, you are right. I forgot to include the rbx/ directory in the make file

Commit:
Update LSP-robloxluau

This release contains the latest version of the RobloxLuau language server.
Sublime-Text-Version-Range: >=4070

alihsaas committed:
Update repository.json

alihsaas committed:
Merge pull request #1 from alihsaas/create-pull-request/patch-rb71yh2

Update LSP-robloxluau

rwols commented:
I wonder if this isn't better some by a few "interface" files, in the same sense of Python interface files (.pyi). For example there's also "love2d" Lua and there's "World of Warcraft" Lua, and so on.

Relevant discussions:

- https://github.com/sumneko/lua-language-server/issues/409
- https://github.com/sumneko/lua-language-server/discussions/389

alihsaas committed:
Update repository.json

alihsaas committed:
Merge branch 'main' into main

alihsaas committed:
Update repository.json

alihsaas committed:
Update repository.json

alihsaas committed:
Update repository.json

alihsaas commented:
> 
> 
> I wonder if this isn't better solved by a few "interface" files, in the same sense of Python interface files (.pyi). For example there's also "love2d" Lua and there's "World of Warcraft" Lua, and so on.
> 
> Relevant discussions:
> 
>     * [Automatically adapt to the project environment sumneko/lua-language-server#409](https://github.com/sumneko/lua-language-server/issues/409)
> 
>     * [Meta files sumneko/lua-language-server#389](https://github.com/sumneko/lua-language-server/discussions/389)

Not sure will this work considering the LSP needs to resolve special syntax which Roblox adds, like the compound assignment and type syntax, these files are just for function/object definitions right?

rchl commented:
Is there still interest to get it into Package Control?
If it works the way it's done then creating per-platform releases isn't really a must.
